spec/system/user_articles_spec.rb

```diff
@@ -0,0 +1,15 @@
+require "rails_helper"
+
+RSpec.describe "User views their articles" do
+  it "displays a list of articles created by the user" do
+    user = create(:user)
+    visit root_path(as: user)
+    3.times { create(:article, user: user) }
+    visit articles_path
+
+    user.articles.each do |article|
+      expect(page).to have_content(article.title)
+      expect(page).to have_content(article.content)
+    end
```

Comment:
**suggestion (blocking):** Rewrite this spec.

How do you feel about the following?

```ruby
create :article, user: user, title: "TEST_ARTICLE_1"
create :article, user: user, title: "TEST_ARTICLE_2"
create :article, title: "TEST_ARTICLE_3"

visit articles_path

expect(page).to have_content "TEST_ARTICLE_1"
expect(page).to have_content "TEST_ARTICLE_2"
expect(page).not_to have_content "TEST_ARTICLE_3"
```

It pays to be more explicit with these things, and you should avoid loops in your specs. They only make the reader try and figure out what is happening. Instead, we should make it as straightforward as possible.

Also, you want to test what we hide from the user. I added a test to ensure they only see articles that belong to them.

What do you think?

Reply:
Addressed in PR #49